### PR TITLE
python3Packages.nunavut: 0.6.2 -> 1.0.1

### PR DESCRIPTION
**Pull request by rmcgibbo**
###### Motivation for this change
Fix `python3Packages.nunavut` (which was broken on hydra) by upgrading `python3Packages.pydsdl`.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

------

Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>python37Packages.nunavut</li>
    <li>python37Packages.pydsdl</li>
    <li>python37Packages.pyuavcan</li>
    <li>python38Packages.nunavut</li>
    <li>python38Packages.pydsdl</li>
    <li>python38Packages.pyuavcan</li>
    <li>python39Packages.nunavut</li>
    <li>python39Packages.pydsdl</li>
    <li>python39Packages.pyuavcan</li>
  </ul>
</details>

### CODE REVIEW

File: pkgs/development/python-modules/pydsdl/default.nix [hunk 1]
```diff
@@ -2,25 +2,26 @@
...
     owner = "UAVCAN";
     repo = pname;
     rev = version;
-    sha256 = "03kbpzdrjzj5vpgz5rhc110pm1axdn3ynv88b42zq6iyab4k8k1x";
+    sha256 = "1hmmc4sg6dckbx2ghcjpi74yprapa6lkxxzy0h446mvyngp0kwfv";
   };
 
-  propagatedBuildInputs = [
-  ];
+  propagatedBuildInputs = [ ];
```
Comment (fabaff) [inline pkgs/development/python-modules/pydsdl/default.nix:15]: Remove this please.

File: pkgs/development/python-modules/nunavut/default.nix [hunk 2]
```diff
@@ -1,13 +1,19 @@
-{ lib, buildPythonPackage, pythonOlder, fetchPypi, pydsdl }:
+{ lib
+, buildPythonPackage
+, pythonOlder
+, fetchPypi
+, pydsdl
+, pytestCheckHook
```
Comment (fabaff) [inline pkgs/development/python-modules/nunavut/default.nix:6]: Not needed as no tests run.

File: pkgs/development/python-modules/nunavut/default.nix [hunk 3]
```diff
@@ -19,7 +25,9 @@
     export HOME=$TMPDIR
   '';
 
-  # repo doesn't contain tests, ensure imports aren't broken
+  # No tests in pypy package and no git tags yet for release versions, see
+  # https://github.com/UAVCAN/nunavut/issues/182
+
```
Comment (fabaff) [inline pkgs/development/python-modules/nunavut/default.nix:30]: Add

```suggestion
  doCheck = false;
```

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112072` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
9 packages built:
</summary>
<ul>
<li>python37Packages.nunavut</li>
<li>python37Packages.pydsdl</li>
<li>python37Packages.pyuavcan</li>
<li>python38Packages.nunavut</li>
<li>python38Packages.pydsdl</li>
<li>python38Packages.pyuavcan</li>
<li>python39Packages.nunavut</li>
<li>python39Packages.pydsdl</li>
<li>python39Packages.pyuavcan</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112072` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
9 packages built:
</summary>
<ul>
<li>python37Packages.nunavut</li>
<li>python37Packages.pydsdl</li>
<li>python37Packages.pyuavcan</li>
<li>python38Packages.nunavut</li>
<li>python38Packages.pydsdl</li>
<li>python38Packages.pyuavcan</li>
<li>python39Packages.nunavut</li>
<li>python39Packages.pydsdl</li>
<li>python39Packages.pyuavcan</li>
</ul>
</details>

**Pull request merged**